GUIDELINES.md

```diff
@@ -0,0 +1,135 @@
+# sweetest test development guidelines
+
+After some time working with sweetest we came up to the conclusion that it leaves a lot of freedom to design tests as we want. But on the other side there is an increasing need for alignment. These guidelines are here for reaching an appropriate level of alignment. Feel free to challenge the current state and to contribute!
+
+## Content
+
+* [Goals](#goals)
+* [One-by-one steps](#one-by-one-steps)
+* [Principles](#principles)
+* [Links](#links)
+
+## Goals
+
+* Put a **layer of abstraction** on the system under test: we call these abstractions _steps_ (as derived from Cucumber)
+  * so if the system under test changes, the test system doesn't need to change
+  * so the test just tells _what_ is tested, not _how_ (tests become more business-centric, whenever possible, all the details are in the steps)
+  * so you can reuse test code
+  * to sum up, there is a proper separation of concerns (separation of business rules, workflow and technical implementation) and the ability to reuse test code
+* **Simplify dependency tree creation** by using configuration and automatic dependency resolution
+* That **simplifies test setup** which makes it **cheaper to have bigger integration tests**
+* With that in mind it becomes worth to **invest more in integration tests** instead of unit tests
+  * which **reduces the use of mocks** and
+  * leads to **more realistic tests**
+
+## One-by-one steps
+
+* Create a module testing configuration in the test sources for each module in your project (e.g. for a module `app` create a file `AppModuleTestingConfiguration` with the following code: `val appModuleTestingConfiguration = moduleTestingConfiguration { ... }`; see below for more info on how to add dependencies to the configuration)
+  * Always use the same package structure for test sources, e.g. if you app resides in a module `app` and has a base package `org.myorg.product.app` then put the module test configuration in exactly in the same package
+  * If there is a class `org.myorg.product.app.view.LoginViewModel` then the test class `LoginViewModelTest` and `LoginViewModelSteps` would reside in the exact same package, too
+  * If you have a dedicated module for test sources you should still stick to the same package structure, e.g. if the module is `:app:test` _don't_ add `test` to the package structure for test sources
+  * If modules depend on each other the module testing configuration should do the same, e.g. `val appModuleTestingConfiguration = moduleTestingConfiguration(dependentConfig1, dependentConfig2, ...) { ... }`, this automatically imports all dependent configurations
+* Create a test class for the _feature under test_ (e.g. `LoginTest`), so the test is agnostic of the underlying structures (classes, models, views, data structures, ...)
```

Comment:
I think this is important to underline: sweetest makes sense for a _feature under tests_. For some simple classes, plain old simple unit tests are the way to go (IMO).

Reply:
i added `Don't use sweetest if it makes no sense (e.g. classes which will never be tested in an integrated fashion)` and `It is not the goal to overcomplicate tests`